190612 Eye tracking docs update - part 1.

sostel opened the pull request:
## Overview
Started updating the MRTK eye tracking docs. Mostly focused on the following docs today:
- EyeTracking_Main.md
- EyeTracking_ExamplesOverview.md
- EyeTracking_TargetSelection.md
- EyeTracking_EyeGazeProvider.md

## Changes
- Numerous changes ranging from fixing typos, reformatting, updating screenshots, etc.

wiwei commented on Documentation/EyeTracking/EyeTracking_ExamplesOverview.md:
```diff
@@ -1,41 +1,80 @@
+In the following, we'll go into more detail what each of the different samples in the [eye tracking example package](https://github.com/Microsoft/MixedRealityToolkit-Unity/tree/mrtk_release/Assets/MixedRealityToolkit.Examples/Demos/EyeTracking) includes:
 
-## Overview of MRTK Eye Tracking Samples
+![List of eye tracking scenes](../Images/EyeTracking/mrtk_et_list_et_scenes.jpg)
```
When looking at the preview, some of these images aren't being loaded (they look to be broken). I think that this is just an artifact of how github previews work, but it may be worth looking at these changes in a local docfx build.

Just to verify that the images and layout are all showing as expected:
https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/DevDocGuide.html#building-docfx-locally

Basically you can see what it will look like for real once it's on the github.io pages. If you need any help with those steps lemme know.

wiwei replied:
Actually, looks like this seems fine - using the diff page apparently isn't a great indication of accuracy, so resolving this one.

sostel replied:
Thanks, Will!

wiwei commented:
/azp run mrtk_docs

wiwei commented on Documentation/EyeTracking/EyeTracking_TargetSelection.md:
```diff
@@ -161,46 +220,132 @@ For the example of the _'smart attentive notifications'_, the following happens:
 allows for quickly assigning different ways to invoke a selection:
 
-- _Pinch gesture_: Setting the 'Select Action' to 'Select' uses the default hand gesture to trigger the selection. This means that the user can simply raise their hand and pinch their thumb and index finger together to confirm the selection.
+- _Pinch gesture_: Setting the 'Select Action' to 'Select' uses the default hand gesture to trigger the selection. 
```
Do you know if this is only possible with the ET examples because of how they disable the far rays?

wiwei replied:
I think my followup question here is around providing a section that explains to users how they modified the default profile or HL2 profile to remove the far interactions that would allow for this to work, and then to provide guidance on when this might be appropriate (or when it wouldn't be).

i.e. I think this involves deleting the default controller from the articulated hand controller type, and if so, it might be good to explicitly show people how to do this.

sostel replied:
Yeah, I'm still trying to figure out what the best flow for explaining it all is. But yep... we should definitely add some info on how to set up the profiles to make it work as expected.

wiwei commented on Documentation/EyeTracking/EyeTracking_TargetSelection.md:
```diff
@@ -2,25 +2,50 @@
+There might be instances where you want to be more specific which type of focus pointers can trigger certain events and allow for simultaneously using multiple far interaction techniques. 
+
+For example:
+In your app, a user can use far hand rays to manipulate some holographic mechanical setup.
```
Reading this section, I'm a little confused above this example, because it sounds like you're not really using both far hand rays and eye focus at the same time - it looks like you're using far hand rays to do some setup, and then you're switching to near interaction and eyes.

The confusing part about this section is that other parts of the docs allude to how having far hands up will turn off eye stuff, but this section is named such that it seems like it would be supported.

wiwei replied:
I guess with a title "Use both eye focus and hand rays at the same time" I would imagine they are both using eyes to look at a checklist AND doing far hand ray manipulation at the same time, but that isn't supported based on current interaction rules.

sostel replied:
It is supported using the EyeTrackingTarget now as it's not based on the FocusHandler anymore. This means that users could grab two holograms with their hand rays and still scroll through text using their eyes, because the auto scroll script is set up to support this unique instance. However, this should not be a default and this is why we didn't simply enable eyes to be supported when hand rays are active, as this could also be confusing if using voice commands and where to channel them to. 
Long story short: Yep, I should explain it better :)

wiwei commented:
@keveleigh in case you're interested in reviewing this.

@sostel can you also fix up the doc build issues:


Build succeeded with warning.
[19-06-13 01:25:34.071]Warning:[BuildCommand.Postprocess.HandlePostProcessorsWithIncremental.HandlePostProcessors.Processing html.ValidateBookmark](Documentation/EyeTracking/EyeTracking_Navigation.md#L14)Illegal link: `<a href="Documentation/EyeTracking/EyeTracking_TargetSelection.md#use-eye-gaze-specific-eyetrackingtarget">EyeTrackingTarget</a>` -- missing bookmark. The file Documentation/EyeTracking/EyeTracking_TargetSelection.md doesn't contain a bookmark named use-eye-gaze-specific-eyetrackingtarget.
	1 Warning(s)

sostel commented:
> @keveleigh in case you're interested in reviewing this.
> 
> @sostel can you also fix up the doc build issues:
> 
> Build succeeded with warning.
> [19-06-13 01:25:34.071]Warning:[BuildCommand.Postprocess.HandlePostProcessorsWithIncremental.HandlePostProcessors.Processing html.ValidateBookmark](Documentation/EyeTracking/EyeTracking_Navigation.md#L14)Illegal link: `<a href="Documentation/EyeTracking/EyeTracking_TargetSelection.md#use-eye-gaze-specific-eyetrackingtarget">EyeTrackingTarget</a>` -- missing bookmark. The file Documentation/EyeTracking/EyeTracking_TargetSelection.md doesn't contain a bookmark named use-eye-gaze-specific-eyetrackingtarget.
> 1 Warning(s)

Hm,

sostel commented:
> @keveleigh in case you're interested in reviewing this.
> 
> @sostel can you also fix up the doc build issues:
> 
> Build succeeded with warning.
> [19-06-13 01:25:34.071]Warning:[BuildCommand.Postprocess.HandlePostProcessorsWithIncremental.HandlePostProcessors.Processing html.ValidateBookmark](Documentation/EyeTracking/EyeTracking_Navigation.md#L14)Illegal link: `<a href="Documentation/EyeTracking/EyeTracking_TargetSelection.md#use-eye-gaze-specific-eyetrackingtarget">EyeTrackingTarget</a>` -- missing bookmark. The file Documentation/EyeTracking/EyeTracking_TargetSelection.md doesn't contain a bookmark named use-eye-gaze-specific-eyetrackingtarget.
> 1 Warning(s)

Ok, this should hopefully be fixed.

wiwei commented:
/azp run mrtk_docs

wiwei commented on Assets/MixedRealityToolkit.Examples/Demos/EyeTracking/General/Scripts/Utils/EyeCalibrationChecker.cs:
```diff
@@ -16,8 +16,11 @@ public class EyeCalibrationChecker : MonoBehaviour
 
-        public UnityEvent OnEyeCalibrationDetected;
-        public UnityEvent OnNoEyeCalibrationDetected;
+        [SerializeField]
```
I don't remember there being code changes as part of these doc updates - did a workspace end up getting overloaded or something?

sostel replied:
Oh, yeah, I guess that got submitted into the incorrect branch. It was just changing public to SerializedField. I can add it to a different PR if you prefer.

wiwei commented on Documentation/EyeTracking/EyeTracking_BasicSetup.md:
```diff
@@ -43,59 +43,83 @@ You can simply select _DefaultMixedRealityToolkitConfigurationProfile_ and then
     - For **Platform(s)** select _'Windows Universal'_.
 
-![MRTK](../Images/EyeTracking/mrtk_setup_eyes_dataprovider.png)
+![MRTK](../Images/EyeTracking/mrtk_setup_eyes_dataprovider.jpg)
```
General question - why the migration from png to jpg?

sostel replied:
I had to update the screenshots anyways and did it for consistency and smaller file sizes.

wiwei commented on Documentation/EyeTracking/EyeTracking_TargetSelection.md:
```diff
@@ -1,26 +1,65 @@
+If the user has his/her hands _not busy_, it would be instinctual to simply touch the check box or select it using a hand ray. 
+However, if the user has his/her hands busy, holding the holographic tools in place, you want to enable the user to seamlessly scroll through the instructions using their eye gaze and simply looking at a check box and say "check it!".
+
+For example:
```
I think that this paragraph got copy pasted from the paragraph above.

wiwei commented on Documentation/EyeTracking/EyeTracking_TargetSelection.md:
````diff
@@ -38,14 +77,16 @@ color when being looked at.
-_MRTK Configuration Profile_ -> _Input System Profile_ -> _Input Actions Profile_. 
+In the example below, we change the color of a hologram by looking at it and pinching or saying "select".
+Which action is required to trigger the event is defined by ```eventData.MixedRealityInputAction == selectAction``` whereby we can set the type of ```selectAction``` in the Unity Editor - by default it's the "Select" action.
+The types of available [MixedRealityInputActions](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Input/InputActions.html) can be configured in the MRTK Profiler via
````
```suggestion
The types of available [MixedRealityInputActions](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Input/InputActions.html) can be configured in the MRTK Profile via
```

wiwei replied:
The weird part is that a profiler is different from a profile in a fairly significant way i.e. perf profiler vs configuration

sostel replied:
Oh, is it? I don't even know. Good catch then!

wiwei commented on Documentation/EyeTracking/EyeTracking_EyesAndHands.md:
```diff
@@ -0,0 +1,20 @@
+## How to support _look + hand motions_ (eye gaze & hand gestures)
```
Consider adding a link to this in the table of contents (toc.yml) so that it's easier for people to find this without clicking one of the 3 other links that are somewhat buried in the other docs.

The same comment applies to the other newly added docs as well - if you believe that they are generally useful even on their own (i.e. person lands on this page and can benefit from it without context from the place they came from in the other docs) it seems good to have this as its own thing.

wiwei replied:
err own thing in toc.yml

sostel replied:
Where can I see the toc.yml as a visitor? I'm just wondering how to test if things look correct.

wiwei commented on Documentation/EyeTracking/EyeTracking_BasicSetup.md:
```diff
@@ -43,59 +43,83 @@ You can simply select _DefaultMixedRealityToolkitConfigurationProfile_ and then
 
-**IMPORTANT:** Don't forget to calibrate HoloLens for the current user. Without calibration, eye tracking won't work. Calibration can be found under system settings.  
+You don't see a _'GazeInput'_ capability? 
+   - Check that your system meets the [prerequisites for using MRTK](../GettingStartedWithTheMRTK.md) (in particular the Windows SDK version). 
```
Can you link directly to the prereqs section:

https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/GettingStartedWithTheMRTK.html#prerequisites

Or more like: (../GettingStartedWithTheMRTK.md#prerequisites)

wiwei replied:
The closer you can get someone to looking at the thing you want them to look at, the easier it will be for them to get the right info.

wiwei commented on Documentation/EyeTracking/EyeTracking_BasicSetup.md:
```diff
@@ -43,59 +43,83 @@ You can simply select _DefaultMixedRealityToolkitConfigurationProfile_ and then
-**IMPORTANT:** Don't forget to calibrate HoloLens for the current user. Without calibration, eye tracking won't work. Calibration can be found under system settings.  
+You don't see a _'GazeInput'_ capability? 
+   - Check that your system meets the [prerequisites for using MRTK](../GettingStartedWithTheMRTK.md) (in particular the Windows SDK version). 
+   - You can also manually add the entry by opening the appxmanifest in an XML editor and adding the following:
```
I *think* this might be required after each Unity build - can you confirm/deny this is the case, and if it is the case (i.e. that this appx manifest change gets blown away on each build) can you add a BOLD note saying that this is required on each Unity build?

sostel replied:
This is only required if you build into a new folder. Otherwise, once you've set it once, you don't have to do it again. I can add a comment about that, but isn't it now outdated anyways, because it's set automatically?

sostel replied:
I added a note about it.

wiwei commented on Documentation/EyeTracking/EyeTracking_BasicSetup.md:
```diff
@@ -43,59 +43,83 @@ You can simply select _DefaultMixedRealityToolkitConfigurationProfile_ and then
+Alternatively you can find the eye calibration in the system settings: Settings -> System -> Utilities -> Open Calibration.
+
+#### Do you see the permission prompt?
+When starting the app on your HoloLens 2 for the first time, a prompt should pop up asking the user for permission to use eye tracking. 
```
Question unrelated to this, but I haven't tested this myself so I'm curious what the answer is - if I accidentally press no, does that apply forever (in which case, i guess I have to go to settings to enable things for the app right?). If that's the case, it might be worth adding that, otherwise people might not know what to do to recover from that very easy accident (i.e. if my index finger tracking is just off and I press no that's bad)

sostel replied:
Yep, it's similar to the microphone privacy settings. You would have to go to the settings page where you see which apps got permissions to use your mic or in this case your eye tracker.

wiwei commented on Documentation/EyeTracking/EyeTracking_BasicSetup.md:
```diff
@@ -43,59 +43,83 @@ You can simply select _DefaultMixedRealityToolkitConfigurationProfile_ and then
 
 ---
-[Back to "Eye Tracking in the MixedRealityToolkit"](EyeTracking_Main.md)
+[Back to "Eye tracking in the MixedRealityToolkit"](EyeTracking_Main.md)
```
Nit: I didn't catch this in a previous review, but I feeeeeeel like the back button in the browser is probably enough for people to get back and that this isn't realllllly necessary

wiwei replied:
Clarification: Super nit

sostel replied:
Yeah, maybe not super necessary, but I also don't like to end up in a dead end when reaching the end of a page.

wiwei commented on Documentation/EyeTracking/EyeTracking_ExamplesOverview.md:
```diff
@@ -1,41 +1,19 @@
-The demo include a number of different use cases for seamlessly combining information about what you are looking at with **Voice** and **Hand** input. 
-This enables users to quickly and effortlessly select and move holograms across their view simply by looking at a target and saying _'Select'_ or performing a hand gesture. 
+# Eye tracking examples in MRTK
+This page covers how to get quickly started with using eye tracking in MRTK by building on our provided [MRTK eye tracking example package](https://github.com/Microsoft/MixedRealityToolkit-Unity/tree/mrtk_release/Assets/MixedRealityToolkit.Examples/Demos/EyeTracking).
```
```suggestion
This page covers how to get quickly started with using eye tracking in MRTK by building on our provided [MRTK eye tracking examples](https://github.com/Microsoft/MixedRealityToolkit-Unity/tree/mrtk_release/Assets/MixedRealityToolkit.Examples/Demos/EyeTracking).
```

wiwei replied:
Mainly because there's no separate package for eye tracking examples, they are part of the examples package.

wiwei replied:
I don't want to lead people down a path of asking "where's the eye tracking package?"

wiwei commented on Documentation/EyeTracking/EyeTracking_ExamplesOverview.md:
```diff
@@ -1,41 +1,19 @@
-## Overview of MRTK Eye Tracking Samples
+We will start with a quick overview of what the individual eye tracking demo scenes are about.
+The individual sample scenes will be [loaded additively](https://docs.unity3d.com/ScriptReference/SceneManagement.LoadSceneMode.Additive.html).
+In case, you wonder what that entails, we will explain further below how to set up and test the MRTK eye tracking demos.
```
Nit: This sentence doesn't really add that much.

Initially I thought that this sentence would tell me more about additive scene loading ("you wonder what this entails") but then the second half of the sentence tells a totally different story (we'll explain how to test the demos!)

It reminds me of something someone told me the other day when they said "today is a cloudy day, as a result I want chocolate"

sostel replied:
It's not completely unrelated. There have been many questions of the additive loading and I wanted to raise it early on that 1) Yes, we do additive scene loading and 2) If you don't know what that means (because a lot of people didn't) then don't worry, we'll get to it. 
But anyways... I rephrased it for now, but it's already outdated anyways with the new scene loading system.

wiwei commented on Documentation/EyeTracking/EyeTracking_ExamplesOverview.md:
```diff
@@ -75,9 +53,83 @@ For more precisely positioning your hologram, you can use additional input from
+
+### 2. Adding scenes to the Build menu
+To load additive scenes during runtime, you must add these scenes to your _Build Settings -> Scenes in Build_ menu first.
+It is important that the root scene is shown as the first scene in the list:
```
Do you know what happens if it's not (i.e. can you mention the dialog that you just added, which will appear if this is violated?).

This can help guide people on the right path (i.e. if we have a section saying here's how you know if something is going wrong, and they may search for the going wrong text, and then be led to this section about how to do things right)

sostel replied:
It's a good point. I leave it like this for now, mainly because this will probably change with the new changes from Lars anyways. I would like to add the revised docs for his change in a different PR though.

wiwei commented on Documentation/EyeTracking/EyeTracking_ExamplesOverview.md:
```diff
@@ -75,9 +53,83 @@ For more precisely positioning your hologram, you can use additional input from
+Simply press down the _right mouse button_ and move the mouse to change its position. 
+When the cursor is hovering over the gems, you will notice that it will snap to the center of the currently looked at gem. 
+This is a great way to test if events are triggered as expected when _"looking"_ at a target. 
+Please beware that the _simulated eye gaze_ via mouse control is a rather poor supplement to our rapid and unintentional eye movements. 
```
```suggestion
Please beware that this is simulating eye gaze via mouse control is a rather poor supplement to our rapid and unintentional eye movements. 
```

wiwei commented on Documentation/EyeTracking/EyeTracking_ExamplesOverview.md:
```diff
@@ -75,9 +53,83 @@ For more precisely positioning your hologram, you can use additional input from
+It is great for testing the basic functionality though before iterating on the design by deploying it to the HoloLens 2 device.
+Coming back to our eye tracking sample scene: The gem rotates as long as being looked at and can be destroyed by "looking" at it and ...
+- Pressing _Enter_ (which simulates saying "select")
+- Actually saying _"select"_ into your microphone
```
```suggestion
- Saying _"select"_ into your microphone
```

wiwei commented on Documentation/EyeTracking/EyeTracking_EyesAndHands.md:
```diff
@@ -0,0 +1,20 @@
+## How to support _look + hand motions_ (eye gaze & hand gestures)
+
+This page explains how to use eye targeting as a primary pointer in combination with hand motions.
+In our [MRTK eye tracking demos](), we describe several examples for using eyes + hands, for example:
```
I think this is missing a link (i.e. () is empty)

sostel replied:
Great catch!

wiwei commented on Documentation/EyeTracking/EyeTracking_IsUserCalibrated.md:
````diff
@@ -0,0 +1,91 @@
+### How to detect the eye calibration state 
+
+The [MixedRealityToolkit.InputSystem.EyeGazeProvider](EyeTracking_EyeGazeProvider.md) provides a ```bool?``` property called ```IsEyeGazeValid```.
+It returns null if no information from the eye tracker is available yet.
````
A quick note - I see a few places where we describe this (which is good!) but also this sets is up to get out of sync if we change this in the future - it would be easy to get one place fixed up but also not others.

This is a general problem when writing docs/guidance, and it's useful to have a single point of reference. This could either be solved by:

1) Referencing a specific definition/API page
2) Linking (xref-ing) to the the generated docs themselves which I believe also have this comment.

Anyways this one is less critical, maybe more of a thing to think about in the future (these are just things I've seen happen in the past, where docs start out great but if things get duplicated in multiple places it is super easy for us to miss updates in a specific location)

sostel replied:
Agreed. Can you point me to instructions on how to use the xref and API refs correctly? I'd like to be able to test it before submitting, but not sure how.

wiwei replied:
I'll have to update the docs to show how to do this - let's follow up on this one afterward.

wiwei commented on Documentation/EyeTracking/EyeTracking_IsUserCalibrated.md:
```diff
@@ -0,0 +1,91 @@
+2. Load _EyeTrackingDemo-00-RootScene.unity_ scene
+
+3. Check out _EyeCalibrationChecker_:
+	- In this scene, we have already a sample for detecting whether the current user is calibrated under the *_EyeCalibrationChecker_ game object*. 
```
Nit: Tabs -> Spaces (I guess maybe do a find replace)

wiwei commented on Documentation/EyeTracking/EyeTracking_Main.md:
```diff
@@ -1,10 +1,16 @@
+_HoloLens 2_ offers an exciting and powerful new input: Eye tracking!
+Eye tracking enables users to quickly and effortlessly engage with holograms across their view and can make your system smarter by better identifying a user's intention. Check out Microsoft's Mixed Reality [documentation on eye tracking on HoloLens 2](https://docs.microsoft.com/en-us/windows/mixed-reality/eye-tracking) for more details, such as explaining powerful applications and design guidelin ...
+
+New to Eye Tracking? No problem! We have created a number of videos, tutorials and samples to get you started in the [Mixed Reality Toolkit](https://github.com/Microsoft/MixedRealityToolkit-Unity) (MRTK)! 
```
```suggestion
New to Eye Tracking? No problem! We have created a number of videos, tutorials and samples to get you started in the [Mixed Reality Toolkit](https://github.com/Microsoft/MixedRealityToolkit-Unity)! 
```

wiwei commented on Documentation/EyeTracking/EyeTracking_Navigation.md:
```diff
@@ -10,9 +10,7 @@ Another interesting application is the hands-free observation of 3D holograms by
-[set up Eye Tracking in your MRTK scene](EyeTracking_BasicSetup.md) and with the basics of accessing Eye Tracking data in Unity including the 
-[EyeTrackingTarget](EyeTracking_TargetSelection.md#use-eye-gaze-specific-eyetrackingtarget) 
-component.
+[set up Eye Tracking in your MRTK scene](EyeTracking_BasicSetup.md) and with the basics of [accessing eye tracking data](EyeTracking_TargetSelection.md) in MRTK Unity.
```
```suggestion
[set up eye tracking in your MRTK scene](EyeTracking_BasicSetup.md) and with the basics of [accessing eye tracking data](EyeTracking_TargetSelection.md) in MRTK Unity.
```

wiwei commented on Documentation/EyeTracking/EyeTracking_Main.md:
```diff
@@ -1,10 +1,16 @@
+_HoloLens 2_ offers an exciting and powerful new input: Eye tracking!
+Eye tracking enables users to quickly and effortlessly engage with holograms across their view and can make your system smarter by better identifying a user's intention. Check out Microsoft's Mixed Reality [documentation on eye tracking on HoloLens 2](https://docs.microsoft.com/en-us/windows/mixed-reality/eye-tracking) for more details, such as explaining powerful applications and design guidelin ...
+
+New to Eye Tracking? No problem! We have created a number of videos, tutorials and samples to get you started in the [Mixed Reality Toolkit](https://github.com/Microsoft/MixedRealityToolkit-Unity) (MRTK)! 
```
```suggestion
New to eye tracking? No problem! We have created a number of videos, tutorials and samples to get you started in the [Mixed Reality Toolkit](https://github.com/Microsoft/MixedRealityToolkit-Unity) (MRTK)! 
```

wiwei replied:
There are a few more instances - you might want to do a 

find . -type f -exec sed -i 's/Eye Tracking/eye tracking/g' {} +

(If you have git bash). If you're not using git bash then there's something else probably.

Obviously do this only with a fresh workspace (I don't know what other places actually don't need this treatment applied)

sostel replied:
Yeah, it's a bit tricky, because sometimes it's at the beginning of a sentence and the Eye is uppercase. So, don't want to just blindly replace. Hope I got them all now though.

wiwei commented on Documentation/EyeTracking/EyeTracking_EyesAndHands.md:
```diff
@@ -0,0 +1,20 @@
+To disable the hand ray pointer, simply remove the _'DefaultControllerPointer'_ in your _Input -> Pointer_ MRTK configuration setting.
+To use eyes and hands as described above in your app, please also make sure that you meet all of the [requirements for using eye tracking](EyeTracking_BasicSetup.md).
+
+![How to remove the hand ray](../Images/EyeTracking/mrtk_setup_removehandray.jpg)
```
Can you add an anchor here (with a sub heading) so that you can link to it directly in places that tell people to "do the profile set up to remove far hand rays")

sostel replied:
Done.

wiwei commented on Documentation/EyeTracking/EyeTracking_TargetSelection.md:
```diff
@@ -1,26 +1,65 @@
+- Look & Say _"Explode"_ or _"Pop"_ (custom voice commands)
+- Look & Bluetooth button
+- Look & Pinch (i.e., hold up your hand in front of you and bring your thumb and index finger together)
+    - _Please note that for this to work, the [hand rays need to be disabled](EyeTracking_EyesAndHands.md)!_
```
Following up on another comment I had above, please have this directly link to the section describing how to do this - otherwise people might not find the right section/information.

sostel replied:
Again - the page is pretty short, but yes, I updated it to the following now: EyeTracking_EyesAndHands.md#how-to-disable-the-hand-ray.

wiwei commented on Documentation/EyeTracking/EyeTracking_TargetSelection.md:
```diff
@@ -1,26 +1,65 @@
+Please note that if hand rays are enabled, the head or eye gaze focus pointer are disabled as soon as the hands come into view.
+
+**IMPORTANT:** Please note that if hand rays are enabled, the head or eye gaze focus pointer are disabled as soon as the hands come into view.
+If you want to support a [_'look and pinch'_ interaction, you need to disable the hand ray](EyeTracking_EyesAndHands.md).
```
Same comment as above on providing a direct link to the hand ray disable guidance section - don't just link to the general page - link to the specific place :)

sostel replied:
Did you click on the page? It consists of 1 title and 2 paragraphs 🤨

sostel replied:
Well, alright. I added more titles to the one-pager and added a link here: EyeTracking_EyesAndHands.md#how-to-disable-the-hand-ray

wiwei commented:
Okay I'm done with this round of feedback, please ping me when I should look again!

sostel commented:
> Okay I'm done with this round of feedback, please ping me when I should look again!

Cool, thanks so much for your diligent review, @wiwei! I think I've addressed most of your feedback. The only open ones are:
- toc.yml: What to add there. Maybe we can do that later?
- xref/API calls: Best practices guide?
- Example for error for additive scene loading: I didn't want to spend too much time on it, as I may have to rewrite it soon with respect to the new scene loading system anyways.

wiwei commented:
/azp run mrtk_docs
